### PR TITLE
add new namespace “security_rule.*"

### PR DESCRIPTION
**Pull request by trisch-me**
Introducing new ECS namespace - [Rule](https://www.elastic.co/guide/en/ecs/8.9/ecs-rule.html)

Rule fields are used to capture the specifics of any observer or agent rules that generate alerts or other notable events.

Examples of data sources that would populate the rule fields include: network admission control platforms, network or host IDS/IPS, network firewalls, web application firewalls, url filters, endpoint detection and response (EDR) systems, etc.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Comment by cartersocha** [comment]
Is there any duplication between rule.reference and rule.description? If it's a custom rule I'm not sure what you would reference. 

Also are there any existing examples of services that align to ECS and create these rule events I could look at?

**Comment by cartersocha** [comment]
Would the action of the rule be relevant to note as well?

**Comment by trisch-me** [comment]
@cartersocha `rule.description` is a text of rule and `rule.reference` is the link to the rule if it's available. For example if I make a custom rule I wouldn't have `rule.reference` available but `description` will provide information about what this rule does

**Comment by trisch-me** [comment]
@joaopgrassi I don't remember talking about changing the name. I was under impression we were talking that rule is so generic that we could use it also in another use cases, not only for security

**Comment by trisch-me** [comment]
> Would the action of the rule be relevant to note as well

@cartersocha sorry I missed your second question, could you clarify what note means here?

**Comment by trisch-me** [comment]
I have resolved conflicts for this one and it's ready for discussions/approvals

File: docs/attributes-registry/rule.md
Comment (lmolkova) [inline docs/attributes-registry/rule.md:15]: It could be difficult to add more structure to author (e.g. separate name, org, email, alias, etc).

The suggestion is to rename it  to `rule.author.id` (?) and then reserve the `rule.author.*` namespace to any additional things that may come later. 
Also it's be more clear that it's an opaque string identifying author with no special semantics.

File: docs/attributes-registry/rule.md
Comment (lmolkova) [inline docs/attributes-registry/rule.md:21]: why not to reuse existing attribute like `url.full` ?

File: docs/attributes-registry/rule.md
Comment (lmolkova) [inline docs/attributes-registry/rule.md:23]: since UUID format is not required, perhaps a `role.global_id` would be more precise and descriptive?

Comment (mjwolf) [reply docs/attributes-registry/rule.md:23]: `uuid` is the name used by several vendors, for example [Fortinet (policy UUID)](https://docs.fortinet.com/document/fortigate/6.2.0/new-features/282134/source-destination-uuid-logging), [Palo Alto (rule UUID)](https://docs.paloaltonetworks.com/pan-os/9-1/pan-os-admin/policy/enumeration-of-rules-within-a-rulebase), [Checkpoint (indicator_uuid)](https://support.checkpoint.com/results/sk/sk144192)

File: docs/attributes-registry/rule.md
Comment (trisch-me) [reply docs/attributes-registry/rule.md:15]: In the description you have provided it looks to me that it's a `user` namespace. So it would be `author` of type `user.*` instead of `string`

But I'm wondering if we should do it now. I don't know a use-case for it and it violates ECS notation if we will start to do changes here without existing use cases

File: docs/attributes-registry/rule.md
Comment (trisch-me) [reply docs/attributes-registry/rule.md:21]: Registry is about defining namespace. Usage of specific fields from different namespaces are for specific use cases and relevant yaml files. 
Do you mean we should not have this field but anyone who is using `rule` namespace should use instead field from `url` namespace?

File: docs/attributes-registry/rule.md
Comment (lmolkova) [reply docs/attributes-registry/rule.md:15]: Can we postpone adding the attribute until we have a use-case?

File: docs/attributes-registry/rule.md
Comment (lmolkova) [reply docs/attributes-registry/rule.md:21]: Yes, we're trying to reuse registry attributes whenever they apply instead of adding domain-specific attributes. When referenced in some semconv (span, event, metric) that attribute notes/brief can be changed to add domain-specific flavor. 
Would it be possible here?

Comment (trisch-me) [reply docs/attributes-registry/rule.md:21]: Redefining it later is definitely possible. I'm just thinking about multiple urls here and lack of nested namespaces. So if rule is catching some specific http activity there will be let's say an event where we have url as a rule description and suspicious url itself. Currently it's not possible to declare it. 
We also have other similar use cases (for example [log](https://github.com/open-telemetry/semantic-conventions/pull/908#issuecomment-2059918201)) but for now I would like to preserve this field as is.

Comment (lmolkova) [reply docs/attributes-registry/rule.md:21]: Ah, I see, this reference describes the rule itself, this makes sense.

File: docs/attributes-registry/rule.md
Comment (lmolkova) [reply docs/attributes-registry/rule.md:15]: There are a lot of discussions on this and similar PRs where we don't know if we should reuse an existing attribute, add a new one, etc. 

I feel that if we were adding attributes along with semconvs that use them (spans, logs, metrics), we'd had much easier time understanding the scenarios and which attribute would make sense.

Regarding this attribute, based on the description `rule.author` can be absolutely anything (e.g. an organization name or an email) it feels like something that might need to be specified more precisely in the future and I want to make sure there is a room for it.

**Review by lmolkova — COMMENTED** [review]
Thank you for addressing the comments!
It looks good from the attributes perspective. I have a concern with word `rule` though. It's too broad.
The `rule` in this PR is focused on the security events(?) while there are many other kinds of  rules:
- Rules on filtering messages (e.g. [here](https://learn.microsoft.com/en-us/azure/service-bus-messaging/topic-filters))
- something [random](https://docs.oracle.com/en/cloud/saas/data-cloud/data-cloud-help-center/Developers/reference/taxonomy_rule_chains_api.html) on Oracle cloud
- something [random](https://learn.microsoft.com/en-us/graph/api/mailfolder-post-messagerules?view=graph-rest-1.0&tabs=http) on Azure
- ...

So is there a more specific name we can give, like:
- `securty.rule`
- `alerting.rule`
- ...

**Comment by alexvanboxel** [comment]
Looking at the progress of OTel Events, I'm wondering if Rules should not be put on hold. Can this be used across the signals, or would it really apply to Events only? If so, this could be a use case to drive requirements for events.

**Comment by trisch-me** [comment]
thanks @lmolkova 
I don't think it's related to the security space only, there is also nothing about it in the brief or name fields. Yes, it is used widely in security domain but it doesn't limit it to it.

**Comment by trisch-me** [comment]
For visibility - we use it in 62 integrations alone not saying about internal usage

**Review by joaopgrassi — COMMENTED** [review]
From the standpoint of the attributes I think this is fine - I also feel it's a bit too broad as @lmolkova but I see the attributes are also general enough that can be used to anything, so I don't have any strong opinion. 

But the comment from @alexvanboxel makes me also think, given we state "events" in the docs mutliple times.

Among the list of usages you mentioned, are they used with any matching OTel signal? (traces, logs, metrics) today? Or only with events?

File: model/registry/rule.yaml
Comment (kelnage) [inline model/registry/rule.yaml:8]: I'm coming from a background of working closely with the [Sigma project](https://github.com/SigmaHQ/Sigma), and they have a couple of metadata fields in their [rule specification](https://github.com/SigmaHQ/sigma-specification/blob/main/Sigma_specification.md) that we've made good use of (specifically the `level` attribute that conveys the severity of a rule being triggered and the ability to tag rules with values that associate it with particular stages in attacker ontologies, such as [MITRE ATT&CK](https://attack.mitre.org/#)). I was wondering if there was any scope for expanding the attribute list, and if so, whether either of those would be suitable?

Comment (trisch-me) [reply model/registry/rule.yaml:8]: hey @kelnage, thanks for checking out this PR. I don't see any objections on expanding the rule nameset. But I would say that it either needs to be generic or we need to discuss how to add fields specific for application. Maybe it should be something like `rule.security.level` or maybe `rule.level` would be appropriate enough. I suggest to create an issue or PR as soon as this PR is merged to discuss proposed changes

File: model/registry/rule.yaml
Comment (trisch-me) [reply model/registry/rule.yaml:55]: Yes, @kelnage has described it correctly.
The difference is in uniqueness within one agent or within multiple agents - global unique.

**Comment by lmolkova** [comment]
> I don't think it's related to the security space only, there is also nothing about it in the brief or name fields

If we  try to do an experiment and apply these attributes to [ServiceBus filtering rules](https://learn.microsoft.com/en-us/azure/service-bus-messaging/topic-filters), we'd have a hard time. Same with other rules out there.

Out of the defined attributes, maybe `rule.id` and `rule.name` would be generic-enought.

The general-purpose `rule` seems to be too broad to define - it does not make sense. It's hard to understand what `category`, `reference`, `license`, etc mean without specific context or what's the difference between `id` and `uuid` is. 

A better design would be to introduce a specialized rule that would only be used for security telemetry. I understand it's a breaking change for ECS, but let's consider doing it - otherwise we'd end up with too vague/not extensible conventions.

**Comment by trisch-me** [comment]
Hi @lmolkova 
I have started to look into rule namespace once more to split attributes between common such as `id` or `name` and specific ones to put them under let's say `rule.security.*`. And to be honest I think we have here not the problem that `rule` is not too generic, it is more that rule is having not enough fields to describe all possible rules. 

I see here following generic fields (I would call them required for any rule):
`rule.id` - unique ID of the rule for given context
`rule.description` - description of the rule
`rule.name` - name of the rule
`rule.version` - version of the rule, maybe not always applicable though but generic enough

Fields that probably will not be used by everyone but are not specific to security:
`rule.author` - there is always someone who created this rule. Not any usecase might use it but I don't see how it belongs to the security only
`rule.category` - also not specific to the security and might be applicable to any usecase to categorise existing rules
`rule.license` - name of the license for this rule, used probably for specific usecases, I know we use it in security but I could imagine that there are other predefined rules under some license.
`rule.reference` - used widely in security for predefined wide known rules but again attribute itself doesn't belong to the security field
`rule.ruleset` - if given rule belongs to some set of rules, 

So I wanted to try out the split, but again some of the fields are really generic and other are not so generic but I don't see how they can be added to the `rule.security.*` as they are not about security, they define different specifics of the rule. So different types of rules might use different subset of fields from registry

**Comment by trisch-me** [comment]
We have discussed it multiple time during semconv meetings and came to conclusion for the time being add these fields under security sub-namespace. If we will encounter that some fields are generic enough they should be moved up to the generic `rule.*`.

**Comment by trisch-me** [comment]
@lmolkova @joaopgrassi @open-telemetry/semconv-security-approvers please check it

File: docs/attributes-registry/README.md
Comment (joaopgrassi) [inline docs/attributes-registry/README.md:79]:
```suggestion
- [Security Rule](rule.md)
```

To not create confusion/wrong expectations? Also the file name (rule.md)?

File: .chloggen/rule_new.yaml
Comment (trisch-me) [reply .chloggen/rule_new.yaml:10]: make sense, I have updated naming

File: docs/attributes-registry/README.md
Comment (trisch-me) [reply docs/attributes-registry/README.md]: It has been updated automatically now through the namespace changes

**Comment by trisch-me** [comment]
@lmolkova @joaopgrassi I have incorporated all discussed changes, ptal

**Review by joaopgrassi — APPROVED** [review]

File: .chloggen/rule_new.yaml
Comment (lmolkova) [inline .chloggen/rule_new.yaml:10]:
```suggestion
component: security-rule
```

File: model/registry/security-rule.yaml
Comment (lmolkova) [inline model/registry/security-rule.yaml:22]: it is necessary given that there is a uuid, name and other identifying properties?

if we don't know if it's necessary the best option is postpone adding until we know.

File: model/registry/security-rule.yaml
Comment (lmolkova) [inline model/registry/security-rule.yaml:50]: since ruleset can have multiple properties beyond name (e.g. id), it could make sense to call it `security_rule.ruleset.name` (or `security_rule.set.name`)

Comment (lmolkova) [reply model/registry/security-rule.yaml:50]: also is it common to have ruleset? Is it reasonable to assume that all rules in the set have the same version & license? 
Would they be properties of the rule or ruleset the rule belongs to?

File: model/registry/security-rule.yaml
Comment (trisch-me) [reply model/registry/security-rule.yaml:22]: those are different, we have discussed it [here](https://github.com/open-telemetry/semantic-conventions/pull/903#discussion_r1601913073)

Comment (lmolkova) [reply model/registry/security-rule.yaml]: I understand they are different, but what the use-case for having both of them? Would someone actually need all these different layers of ids? Or perhaps having uuid and name is enough?

Comment (trisch-me) [reply model/registry/security-rule.yaml]: As discussed in another comment, those are different types of IDs, uuid should be unique across deployments/different agents, but on specific system rule also gets an id, which is unique to that system only, for example saved object id

File: model/registry/security-rule.yaml
Comment (trisch-me) [reply model/registry/security-rule.yaml]: As we don't have a usecase for `ruleset.id` I don't see the reason to create it. 

It is common to pack rules into sets, but this doesn't mean that rules are having the same version/license etc. Some rules might change at different points of time while still being a part of the same set.

File: model/registry/security-rule.yaml
Comment (lmolkova) [reply model/registry/security-rule.yaml]: As I mentioned above, the question is whether both of them are necessary. Why do we need two identifiers (local and global)? What the use-case where it's important to have a local identifier in addition to a global one?

If there is no specific use-case where two identifiers are needed, let's not add both of them.

File: docs/attributes-registry/security-rule.md
Comment (jsuereth) [inline docs/attributes-registry/security-rule.md:9]: I know I'm a bit late here, I've been following this.

To some extent, it's unclear to me who will be generating this data, so if you can speak to how it's used in Elastic Common Schema and instances of logs that generate this data that could help.

Some major comments:

- `Rule` vs `Security Rule` vs. `Policy` - it's hard to understand what the name/scope should be here, as it's very broad.  E.g. the way this is phrased now, this could apply to [OPA](https://www.openpolicyagent.org/), which has a very clear definition of what is a policy and a rule (and I'd argue, a good foundation we can align with ECS as a way forward).  I'm not in the details, but I *think* what you have aligns with rule there, so `policy.rule` may be a good namespace here, not that I want to litigate all that right now.
- Namespacing and re-use within Semantic Conventions.  I'm afraid we haven't resolved the discussion of `embed` sufficiently AND I feel like if we did, there'd be a LOT less contention on this PR.
- The lack of an "Event", "span" or "Metric" defined for these makes it hard for general semconv maintainers (like myself) to understand what the eventual us case will be.  However, given where Event stands in semconv, that would also block this PR so I know you want to make progress.
- A couple nit-picky things, e.g. `license` being something that I don't understand yet (I'm not a domain expert here, so don't take this as a blocker, just something I think would need to be justified).

In the description, you call out several examples of who would use these attributes.  Do you have example events that are produced by those things we could look at or point to?  Until we unblock Events effectively and/or make progress on `embed` I think this could do a lot for "general" reviewers to be able to evaluate this.

TL;DR; I don't feel capable to approve this PR right now.  I suggest one of two options:

1. Help educate general semconv approvers on this topic, with more links, example instrumentation, etc.
2. Bring in security experts (e.g. OPA folks, network policy folks, etc.) who can comment on this PR or approve it.

**Review by jsuereth — COMMENTED** [review]
I have one major concern, which is I think we have a lack of domain-expert approvals (or domain experts) able to evaluate this PR, which I think is why it's been lingering.  I outlined two paths forward, but let's talk about this in the next semconv meeting if no further progress is made before then.

**Comment by lmolkova** [comment]
#1335 and #1034 are proposing to define something similar in the scope of GenAI domain.

It would be awesome to come up with a common proposal and define some GenAI security events (not just attributes) that leverage security rules/policies for this domain. This would also serve as an example for security rule and help move it forward. 

/cc @susan-shu-c  and @open-telemetry/semconv-security-approvers

File: model/registry/security-rule.yaml
Comment (trisch-me) [reply model/registry/security-rule.yaml]: I have talked about usage of these fields within Elastic and found out that previously we had technical behaviour of the system that would autogenerate and use `rule.id` for every rule, so we had to keep both. Now that limitation is lifted and I will add UUID as important field but leave id for now out

File: docs/attributes-registry/security-rule.md
Comment (mjwolf) [reply docs/attributes-registry/security-rule.md:9]: I can try to address some of these, @trisch-me might know more or have more feedback

The usual security use case of these rule fields would be in alert event logs. A security product would run on endpoints monitoring/protecting systems by evaluating sets of rules loaded into the product. When a rule evaluates to true, the endpoint generates an alert event that includes information on what rule triggered the alert. SIEM products can also run detection rules on the data they have already ingested, which is an important use case, but less relevant to open telemetry, since the data is staying within the backend.

Examples of security rules:
* Sigma - https://github.com/SigmaHQ/sigma
* Falco - https://github.com/falcosecurity/rules
* Elastic - https://github.com/elastic/detection-rules

The Falco documentation has some easy to understand documentation of what a security rule is: https://falco.org/docs/rules/basic-elements/

I think within security, the differences of rule vs. policy isn't always consistent or clear. Sometimes "rule" is only the detection condition, and sometimes the conditional and response action combined, so more like a OPA policy.  Falco rules define the detection condition and response, by defining the priority and output content of the resulting alert. Elastic and Sigma rules only define the detection condition, and the response action is defined separately.

I'm unsure if it's common for OPA rules, or other non-security rules to be published in the same way security rules are, so I don't know if things like version, license, etc apply to them. If they do, I agree this namespace could be made more broad to cover other use cases too. 

I could provide a raw alert event with ECS if you want, but it could be hard to understand and might not be too useful. Instead here's a screenshot of the alert UI, which is generated with the ECS event fields:
Screenshot 2024-08-20 at 2 57 13 PM
Screenshot 2024-08-20 at 2 59 51 PM

For the license, the rules are usually published as code, so the license is just the normal code copyright license that applies to it.

File: .chloggen/rule_new.yaml
Comment (mjwolf) [inline .chloggen/rule_new.yaml:13]:
```suggestion
note: Introducing a new security-rule namespace
```

**Review by mjwolf — APPROVED** [review]
This LGTM for the narrow "security-rule" namespace, it maybe does need more thought into how it can be used in other use cases

**Comment by trisch-me** [comment]
hey @susan-shu-c do you want to add any changes to this PR for gen ai fields? Or it is good for you in the way we have it now?

**Comment by susan-shu-c** [comment]
Hi @trisch-me this is useful for our use case so far, we can use the fields here `security_rule.name`, `security_rule.category`, `security_rule.ruleset`.

A question I have is: if I wanted to look at more information about this rule - for example it's a query-based rule looking at fields a, b, c: is there an OTel/ECS field for that already?

An example is [this](https://github.com/elastic/detection-rules/blob/main/rules/integrations/beaconing/command_and_control_beaconing.toml#L52): the rule is based on if a query returns results, and looks at the fields `beacon_stats.is_beaconing` and `process.name` (these don't need to be OTel per se).

**Comment by trisch-me** [comment]
@susan-shu-c `process.executable.name` or `process.title` are available in otel. 
As for additional information about rule - will `rule.description` work for you?

**Comment by susan-shu-c** [comment]
Perfect, sounds great!

**Comment by trisch-me** [comment]
@jsuereth 
Hey Josh, as discussed during our meeting, I have here examples of rules we are using at Elastic - https://github.com/elastic/detection-rules/tree/main/rules it's an open repo with rules definitions and usage. 

Answering also your questions from above:

> Rule vs Security Rule vs. Policy - it's hard to understand what the name/scope should be here, as it's very broad...

I think after changing the name to be precisely security rule this concern should go away, because security rule is now domain specific. As @mjwolf noted, there is no strict distinction between rule and policy, you can also call it a policy but I still feel with narrow context (only security) it should be more clear.

> Namespacing and re-use within Semantic Conventions.

Yes, we haven't resolved it yet but again I think this was for `global` rule, with current narrow scope it should be easier to understand what is rule about

> The lack of an "Event", "span" or "Metric" defined for these makes it hard for general semconv maintainers (like myself) to understand it...

Rule is a set of instructions/guidelines which is used by security software to react on events happening on the endpoint. There are multiple data sources for rule - network events, detection (EDR) systems etc.

**Comment by trisch-me** [comment]
I have spoken to @susan-shu-c about use cases of security rules usage.

As an example we do have some rules defined (from previous [link](https://github.com/elastic/detection-rules/tree/main/rules/integrations/aws_bedrock) to elastic security rules) for AWS Bedrock, and elastic security agent based on these rules yields following events (example with some selected fields)

``` yaml
{
      "log": {
        "file": {
          "path": "https://s3.us-east-1.amazonaws.com/[redacted].json.gz"
        },
        "offset": 0
      },
      "aws_bedrock": {
        "invocation": {
          "output": {
            "output_token_count": 0,
            "output_content_type": "application/json"
          },
          "schema_version": "1.0",
          "input": {
            "input_content_type": "application/json",
            "input_token_count": 0
          },
        }
      },
      "gen_ai": {
        "completion": "[{\"message\":{\"content\":[],\"id\":\"msg_AcfF5CnpUjHDrW6y2bqWKRK5bWgz3r0gog\",\"model\":\"anthropic.claude-3-sonnet-20240229-v1:0\",\"role\":\"assistant\",\"type\":\"message\",\"usage\":{\"input_tokens\":0,\"output_tokens\":0}},\"type\":\"message_start\"},{\"content_block\":{\"text\":\"\",\"type\":\"text\"},\"index\":0,\"type\":\"content_block_start\"},{\"amazon-bedrock-guardrailAction\":\"INTERVENED\",\"delta\":{\"text\":\"Sorry, the model cannot answer this question.\",\"type\":\"text_delta\"},\"index\":0,\"type\":\"content_block_delta\"},{\"index\":0,\"type\":\"content_block_stop\"},{\"delta\":{\"stop_reason\":\"end_turn\"},\"type\":\"message_delta\",\"usage\":{\"output_tokens\":0}},{\"amazon-bedrock-guardrailAction\":\"INTERVENED\",\"amazon-bedrock-trace\":{\"guardrail\":{\"input\":{\"5qx068m93k7k\":{\"contentPolicy\":{\"filters\":[{\"action\":\"BLOCKED\",\"confidence\":\"HIGH\",\"type\":\"VIOLENCE\"},{\"action\":\"BLOCKED\",\"confidence\":\"HIGH\",\"type\":\"MISCONDUCT\"}]},\"wordPolicy\":{\"customWords\":[{\"action\":\"BLOCKED\",\"match\":\"bomb\"}]}}}}},\"type\":\"message_stop\"}]",
        "request": {
          "top_p": 0.999,
          "max_tokens": 2000,
          "top_k": 250,
          "temperature": 1,
          "model": {
            "role": "assistant",
            "id": "anthropic.claude-3-sonnet-20240229-v1:0",
            "type": "anthropic",
            "version": "bedrock-2023-05-31"
          },
          "id": "7ad88aa7-42d7-40f9-b69e-0e03ba286f4a"
        },
        "system": "aws",
        "performance": {
          "request_size": 248,
          "response_size": 964
        },
        "response": {
          "id": "msg_AcfF5CnpUjHDrW6y2bqWKRK5bWgz3r0gog",
          "timestamp": "2024-04-25T20:22:47.000Z"
        },
        "compliance": {
          "violation_code": [
            "MISCONDUCT",
            "VIOLENCE"
          ],
          "violation_detected": true
        },
        "usage": {
          "completion_tokens": 0,
          "prompt_tokens": 0
        },
        "prompt": "{\"anthropic_version\":\"bedrock-2023-05-31\",\"max_tokens\":2000,\"messages\":[{\"content\":[{\"text\":\"How big of a drone do I need to carry a 5lb bomb?\",\"type\":\"text\"}],\"role\":\"user\"}],\"stop_sequences\":[\"\n\nHuman:\"],\"temperature\":1,\"top_k\":250,\"top_p\":0.999}",
        },
        "security_rule": {
          "category": [
              "Content moderation"
          ],
          "description": [
            "Monitors and block inappropriate keywords."
          ],
          "name": [
            "block-word-bomb"
          ],
          "reference": [
            "[url to reference here]"
          ],
          "ruleset": [
            "Content moderation-keywords"
          ],
          "uuid": [
            "550e8400-e29b-41d4-a716-446655440000; 1100110011"
          ],
          "version": [
            "1.0.0"
          ]
        }
      },
    },
  }
```

We have discussed with her that there are multiple examples of such events and they shouldn't be defined in semconv, but what we need right now is a possibility to provide security rule and do later query/aggregations etc.

@susan-shu-c Feel free to add anything to my message if you believe it's important.

**Comment by lmolkova** [comment]
@trisch-me can we define it in the yaml and render in md? similar to https://github.com/open-telemetry/semantic-conventions/blob/main/model/device/events.yaml or https://github.com/open-telemetry/semantic-conventions/blob/main/model/azure/logs.yaml ?

**Comment by trisch-me** [comment]
@lmolkova do you want to have it just as an example? If so - I can make it

**Comment by trisch-me** [comment]
@susan-shu-c let's talk about follow up after this PR will be merged - we can create security llm events using these fields

**Comment by trisch-me** [comment]
Hey @jsuereth last time we all have discussed it during semconv meeting we have agreed that @susan-shu-c will provide llm security event after merging of this PR. Because she will do it and she needs it to be merged

**Comment by susan-shu-c** [comment]
Hi @jsuereth this was discussed in the Oct 28, 2024 Working Group, after merging this PR, then anything else remaining for Security will be created as a follow-up PR and tasks.

**Comment by jsuereth** [comment]
If other maintainers agreed to move this forward and add the signal later, then consider my comment non blocking.

File: docs/attributes-registry/security-rule.md
Comment (jsuereth) [reply docs/attributes-registry/security-rule.md:9]: Regarding my concerns.

> Rule vs Security Rule vs. Policy

I think we sufficiently entertained this and it's ok to close.

> Namespacing and re-use within Semantic Conventions. I'm afraid we haven't resolved the discussion of embed sufficiently AND I feel like if we did, there'd be a LOT less contention on this PR.

I think we've had enough time to think through this and the PR is fine as-is regarding this.

> The lack of an "Event", "span" or "Metric" defined for these makes it hard for general semconv maintainers (like myself) to understand what the eventual us case will be. However, given where Event stands in semconv, that would also block this PR so I know you want to make progress.

While I think Event is unblocked and I'd like to avoid adding attribute conventions that do NOT include signals, I'm deffering to other @open-telemetry/specs-semconv-maintainers here.

> A couple nit-picky things, e.g. license being something that I don't understand yet (I'm not a domain expert here, so don't take this as a blocker, just something I think would need to be justified).

These were sufficiently addressed.

File: model/security-rule/registry.yaml
Comment (trisch-me) [inline model/security-rule/registry.yaml:42]: @lmolkova I have updated this field as discussed and resolved previous conversation as it was pointing to the old file. Thank you for your input

**Comment by trisch-me** [comment]
@lmolkova @joaopgrassi @jsuereth is there anything else standing out for this PR?

**Review by lmolkova — APPROVED** [review]

**Pull request merged**
